shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -34,43 +34,57 @@ public static string GetTypeDisplayName(object item, bool fullName = true)
             return item == null ? null : GetTypeDisplayName(item.GetType(), fullName);
         }
 
-        public static string GetTypeDisplayName(Type type, bool fullName = true)
+        /// <summary>
+        /// Pretty print a generic type name.
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/>.</param>
+        /// <param name="fullName"><c>true</c> to print a fully qualified name.</param>
+        /// <param name="includeGenericParameterNames"><c>true</c> to include generic parameter names.</param>
+        /// <returns>The pretty printed type name.</returns>
+        public static string GetTypeDisplayName(Type type, bool fullName = true, bool includeGenericParameterNames = false)
         {
             var builder = new StringBuilder();
-            ProcessType(builder, type, fullName);
+            ProcessType(builder, type, new DisplayNameOptions(fullName, includeGenericParameterNames));
             return builder.ToString();
         }
 
-        private static void ProcessType(StringBuilder builder, Type type, bool fullName)
+        private static void ProcessType(StringBuilder builder, Type type, DisplayNameOptions options)
         {
             if (type.IsGenericType)
             {
                 var genericArguments = type.GetGenericArguments();
-                ProcessGenericType(builder, type, genericArguments, genericArguments.Length, fullName);
+                ProcessGenericType(builder, type, genericArguments, genericArguments.Length, options);
             }
             else if (type.IsArray)
             {
-                ProcessArrayType(builder, type, fullName);
+                ProcessArrayType(builder, type, options);
             }
             else if (_builtInTypeNames.TryGetValue(type, out var builtInName))
             {
                 builder.Append(builtInName);
             }
-            else if (!type.IsGenericParameter)
+            else if (type.IsGenericParameter)
```

Comment:
You can invert this condition and add else if to reduce nesting.

Reply:
It just seems harder to reason about at a glance. I'll leave it as is.